test/test_utils.py

```diff
@@ -722,7 +731,96 @@ def test_load_commit_from_forked_repo(self):
+                lines = f.readlines()
+            with open(file_path, "w") as f:
+                for line in lines:
+                    if line.strip("\n") != "ailzhang_torchhub_example":
+                        f.write(line)
+
+        file_path_legacy = os.path.join(torch.hub.get_dir(), 'trusted_list_legacy')
+        if os.path.exists(file_path_legacy):
+            with open(file_path_legacy, "r") as f:
+                lines = f.readlines()
+            with open(file_path_legacy, "w") as f:
+                for line in lines:
+                    if line.strip("\n") != "ailzhang_torchhub_example_master":
+                        f.write(line)
+
+    @retry(Exception, tries=3)
+    @patch('torch.hub._get_trusted_input', return_value='')
+    def test_untrusted_repo(self, input):
+        with self.assertRaisesRegex(
+                Exception,
+                'Untrusted repository.'):
+            model = torch.hub.load(
+                'ailzhang/torchhub_example',
+                'mnist_zip_1_6',
+                force_reload=True,
+                trust_repo=False)
+
+    @retry(Exception, tries=3)
+    @patch('torch.hub._get_trusted_input', return_value='y')
+    def test_trusted_repo(self, input):
+        self._cleanup()
+        model = torch.hub.load(
+            'ailzhang/torchhub_example',
+            'mnist_zip_1_6',
+            force_reload=True,
+            trust_repo=False)
+
+    @retry(Exception, tries=3)
+    @patch('torch.hub._get_trusted_input', return_value='y')
+    def test_check_repo(self, input):
+        self._cleanup()
+        model = torch.hub.load(
+            'ailzhang/torchhub_example',
+            'mnist_zip_1_6',
+            force_reload=True,
+            trust_repo="check")
+
+    @retry(Exception, tries=3)
+    def test_none_repo(self):
+        self._cleanup()
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            model = torch.hub.load(
+                'ailzhang/torchhub_example',
+                'mnist_zip_1_6',
+                force_reload=True,
+                trust_repo=None)
+            assert len(w) == 1
+            assert issubclass(w[-1].category, UserWarning)
+            assert "You are about to download and run code from an untrusted repository" in str(w[-1].message)
+
+    @retry(Exception, tries=3)
+    @patch('torch.hub._get_trusted_input', return_value='n')
+    def test_check_repo_legacy(self, input):
+        self._cleanup()
+        # add repo to legacy
+        file_path_legacy = os.path.join(torch.hub.get_dir(), 'trusted_list_legacy')
+        if not os.path.exists(file_path_legacy):
+            pathlib.Path(file_path_legacy).touch()
+        with open(file_path_legacy, "a") as f:
+            f.write('ailzhang_torchhub_example_master')
+        # even though we choose to check and answer no to the prompt, we should pass through as the repo is in the
+        # trusted list (legacy)
```

Comment:
There should be no prompt, right? I think we should be able to remove the mock entirely and the comment could become

> the repo + branch is in the legacy list so it is considered a trusted repo, and the user isn't prompted anything